.github/workflows/main.yml

```diff
@@ -18,25 +18,30 @@ jobs:
         with:
           python-version: '3.10'
 
-      - name: Install dependencies
-        run: |
-          pip install wheel
-          pip install -r requirements.txt
-          pip install -e .
-
-      - name: Check formatting with Black
-        run: |
-          black --check --exclude "(src|alembic)/" .
-
-      - name: Run headless tests
-        uses: coactions/setup-xvfb@v1
+      - uses: snok/install-poetry@v1
         with:
-          run: python -m pytest
-          working-directory: ./  # Optional: Specify the working directory if needed
-          options:  # Optional: Add any additional options or arguments for pytest
+          version: 1.5.1
+          virtualenvs-create: true
+          virtualenvs-in-project: true
 
-      - name: flake8 Lint
-        uses: py-actions/flake8@v2
+      - name: Cache deps
+        id: cache-deps
+        uses: actions/cache@v2
         with:
-          plugins: "flake8-docstrings"
-          extra-args: "--docstring-convention=google --exclude=alembic/versions"
+          path: .venv
+          key: pydeps-${{ hashFiles('**/poetry.lock') }}
+
+      - run: poetry install --no-interaction --no-root
+        if: steps.cache-deps.outputs.cache-hit != 'true'
+
+      - name: Install Tesseract
+        run: brew install tesseract
```

Comment:
Does it make sense to run the install script here?

Reply:
Currently one of the scrubbing tests in `test_scrub.py` requires to download tesseract in order to pass. Ideally I wouldn't want to download just for one test, for now I added this action.

Reply:
I don't follow. Can you please clarify what is the difference between downloading tesseract in the install script, and running `brew install tesseract`? Isn't that what the install script does? https://github.com/OpenAdaptAI/OpenAdapt/blob/main/install/install_openadapt.sh#L119

Reply:
Yeah it does, I haven't tested this script but I can execute and try it out. :)

Reply:
@abrichr I realized the script also downloads unnecessary stuff like `git` and `brew`, so I think its better I keep `brew install tesseract` but also add the install option for windows.